BayesianOptimization/discussion_parameters.md

```diff
@@ -0,0 +1,46 @@
+
+## Biceps case
+There is the [isometric biceps case](../opendihu_examples/isometric_contraction/biceps_muscle/) for which we want to discuss special parameters. If we use the recommended parameters, we get the following optimization plot: 
+
+![](../figures/isometric_biceps.png)
+
+We have found a maximum this way, which is good, but there are some disadvantages of this optimization process.
+- There are very large areas of high uncertainty in which we don't know if we might have a hidden maximum
+- The mean function is not very smooth and the trials are in local maxima, which is not very realistic
+
+The question now is, if there is a parameter setup that finds the same (or a better) maximum without those disadvantages.
+
+#### Parameters with low impact
+There are a few parameters that do not have a lot of impact on the general result, which we are not going to look at here. They can be adjusted after the major parameter choices for finetuning. Examples for that are the mean function, the noise or the stopping criterion. All these parameters will be chosen as in the general recommendation.
+
+#### Entropy Search
+The recommended setup with the Matérn kernel and $\nu$=0.5 is aparently working fine, but not optimally. If we change $\nu$ to 1.5 or 2.5, the result is immediately much worse. Since $\nu$ is the smoothness parameter, a higher value changes the assumed smoothness of the function. This has two disadvantages. The first one is the compatibility with ES. This acquisition function is working with minimizing entropy, which is much easier if we have more uncertainty. We lose that with a higher smoothness parameter. The second disadvantage is that we might not find the maximum in this case. The maximum is just at the edge of a high jump downwards. With a high smoothness the mean function will interpolate the trials in a way that will not allow high jumps, so we might look for a maximum elsewhere.
+
+However, there is the option of adjusting the ES acquisition function so that we at least remove the large areas of high uncertainty. An option for this would be an evaluation of our function every 3 or 4 evaluations at the value with the highest uncertainty. In the case above this would add about 2 evaluations. That can have two effects. Either this results in a new and better maximum, or we get a nicer picture in the end. Keeping in mind that an evaluation takes an hour, one has to evaluate if this is worth the risk. Since you can always do just one more evaluation after the optimization process is done, the acquisition function should probably not be changed (if we're using ES), and depending on the result we can add more evaluations in areas we deem necessary.
```

Comment:
```suggestion
However, there is the option of adjusting the ES acquisition function so that we at least remove the large areas of high uncertainty. An option for this would be an evaluation of our function every 3 or 4 evaluations at the value with the highest uncertainty. In the case above this would add about 2 evaluations. That can have two effects. Either this results in a new and better maximum, or we get a nicer picture in the end. Keeping in mind that an evaluation takes an hour, one has to evaluate if this is worth the risk. Since you can always do just one more evaluation after the optimization process is done, the acquisition function should probably not be changed (if we are using ES), and depending on the result we can add more evaluations in areas we deem necessary.
```